hazelcast-client/src/main/java/com/hazelcast/client/config/ClientFlakeIdGeneratorConfig.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.util.Preconditions;
+
+/**
+ * The {@code ClientFlakeIdGeneratorConfig} contains the configuration for the client
+ * regarding {@link com.hazelcast.core.HazelcastInstance#getFlakeIdGenerator(String)
+ * Flake ID Generator}.
+ */
+public class ClientFlakeIdGeneratorConfig {
```

Comment:
Seems this is equal to the `FlakeIdGeneratorConfig` so I would remove this class and all occurrences on the client (parsers, tests, ...)

Reply:
Was thinking about it, but i don't think it's a good idea: we might once want to add some setting to member but not to client... And the one on member is serializable, has read-only version etc.
And we certainly cannot remove parsers, they parse different files according to different XSD.

Reply:
I would rather not add classes now for possible and questionable use cases in the indeterminate future.